### PR TITLE
Fix powershell 5 compatibility

### LINKED ISSUE #21968: [vcpkg] appdeploy.ps1 "Join-Path" InvalidArgument error when trying to install plugin dependencies
**Describe the bug**
Powershell error when compiling projects using e.g. `magnum-plugins` port:
```
Join-Path : A positional parameter cannot be found that accepts argument 'magnum'.
At C:\Repos\vcpkg\scripts\buildsystems\msbuild\applocal.ps1:124 char:61
+ ... um $targetBinaryDir (Join-Path "$g_install_root" 'bin' 'magnum') "$_"
+                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Join-Path], ParameterBindingExcepti
   on
    + FullyQualifiedErrorId : PositionalParameterNotFound,Microsoft.PowerShell.Commands
   .JoinPathCommand
```

**Environment**
- OS: Windows
- Compiler: MSVC 19.29.30136.0

**To Reproduce**
1. ./vcpkg install magnum-plugins
2. Set up basic cmake or VS project that uses Magnum::Trade from the `magnum` port (this triggers installation of plugins into a magnum/importers directory)
2. See error during build

**Expected behavior**
No error, plugin dependencies are copied directly.

**Failure logs** 
-(please attached failure logs)

**Additional context**
Commit https://github.com/microsoft/vcpkg/commit/7dff5e821e4a0b7fe43e8a28c51357217bb4bef6 by @dg0yt uses `Join-Path` which joins two paths, but tries to join three.

### PR DESCRIPTION
**Pull request by dg0yt**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #21968

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  --

### CODE REVIEW

File: scripts/buildsystems/msbuild/applocal.ps1 [hunk 1]
```diff
@@ -119,9 +119,9 @@ function resolve([string]$targetBinary) {
             if (Test-Path function:\deployOpenNI2) { deployOpenNI2 $targetBinaryDir "$g_install_root" "$_" }
             if (Test-Path function:\deployPluginsIfMagnum) {
                 if ($g_is_debug) {
-                    deployPluginsIfMagnum $targetBinaryDir (Join-Path "$g_install_root" 'bin' 'magnum-d') "$_"
+                    deployPluginsIfMagnum $targetBinaryDir (Join-Path (Join-Path "$g_install_root" 'bin') 'magnum-d') "$_"
```
Comment (JackBoosY) [inline scripts/buildsystems/msbuild/applocal.ps1:122]: Should we check powershell version first?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: scripts/buildsystems/msbuild/applocal.ps1 (hunk 1 above)
Comment (dg0yt) [reply scripts/buildsystems/msbuild/applocal.ps1:122]: No. Why?

Comment (JackBoosY) [reply scripts/buildsystems/msbuild/applocal.ps1:122]: Ah it's supported in all version. Right.

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**